starky2/src/cross_table_lookups.rs

```diff
@@ -0,0 +1,176 @@
+use plonky2::field::extension_field::{Extendable, FieldExtension};
+use plonky2::field::field_types::Field;
+use plonky2::field::packed_field::PackedField;
+use plonky2::field::polynomial::PolynomialValues;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::challenger::Challenger;
+use plonky2::plonk::config::GenericConfig;
+use plonky2::plonk::plonk_common::reduce_with_powers;
+use plonky2::util::reducing::ReducingFactor;
+
+use crate::all_stark::Table;
+use crate::config::StarkConfig;
+use crate::constraint_consumer::ConstraintConsumer;
+use crate::permutation::PermutationChallenge;
+use crate::stark::Stark;
+use crate::vars::StarkEvaluationVars;
+
+pub struct CrossTableLookup {
+    pub looking_table: Table,
+    pub looking_columns: Vec<usize>,
+    pub looked_table: usize,
```

Comment:
Was there a reason to use Table for looking but usize for looked?

Reply:
Oops, not at all just a typo. Fixed.